src/Processors/Formats/Impl/Parquet/ParquetDataBuffer.h

```diff
@@ -54,11 +54,10 @@ class ParquetDataBuffer
         auto necessary_bytes = count * sizeof(ParquetType);
         checkAvaible(necessary_bytes);
 
-        const ParquetType* src = reinterpret_cast<const ParquetType*>(data);
-
         for (std::size_t i = 0; i < count; i++)
         {
-            dst[i] = static_cast<TValue>(src[i]);
+            auto offset = i * sizeof(ParquetType);
+            dst[i] = unalignedLoad<TValue>(data + offset);
```

Comment:
is it guaranteed that `sizeof(ParquetType) == sizeof(TValue)`. let's add `static_assert` if so.

Reply:
It is guaranteed that `sizeof(ParquetType) <= sizeof(TValue)`

Reply:
is it fine that we don't consider endianness of the machine running this code. afaiu parquet always writes in LE, but what if we're running on a BE?

Reply:
Native reader is only available on LE systems: https://github.com/ClickHouse/ClickHouse/blob/4212fa4210175b5566d7571eb2a4f40379ace550/src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp#L694